### PR TITLE
PIPE-22234 Show error message instead of loading when relevant

### PR DESCRIPTION
**Pull request by vivek-harness**
When a pipeline was created but not saved, and then if it was tried to run, the UI showed indefinite "Loading" state instead of showing the error message.

This issue is now fixed as part of this PR.

https://github.com/user-attachments/assets/f7479d79-add1-40cf-8cc6-365ed427647d

### CODE REVIEW

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx [hunk 1]
```diff
@@ -109,14 +119,18 @@ export default function RunPipelineForm({
...
-        // TODO: error toast here ?
+        setLoading(false)
+        setErrorMessage(error.message)
       })
   }
 
+  if (errorMessage) {
+    return <>{errorMessage}</>
+  }
+
```
Comment (sans-harness) [inline apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx:133]: maybe style the message error message as `text-destructive`?

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx [hunk 2]
```diff
@@ -109,14 +118,18 @@ export default function RunPipelineForm({
...
       .catch(error => {
         console.error(error)
-        // TODO: error toast here ?
+        setLoading(false)
+        setErrorMessage(error.message)
       })
   }
 
+  if (errorMessage) {
+    return <span className="text-red-600">{errorMessage}</span>
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx:130]: use `<Alert><AlertTitle/></Alert>` instead

**Comment by sans-harness** [comment]
The alert should ideally not overlap with the dialog close button.

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx [hunk 3]
```diff
@@ -52,28 +63,41 @@ export default function RunPipelineForm({
...
+                try {
+                  const pipelineObj = parse(decodeGitContent(pipelineFileContent?.content?.data))
+                  setPipeline(pipelineObj)
+                } catch (ex: unknown) {
+                  console.error(ex)
+                  setErrorMessage((ex as UsererrorError)?.message || null)
+                }
+              },
+              ex => {
+                console.error(ex)
```
Comment (sans-harness) [inline apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx:87]: NIT: remove the console.errors if we are handling them with a dialog?

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx (hunk 1 above)
Comment (vivek-harness) [reply apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx]: Used Alert component instead

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx (hunk 2 above)
Comment (vivek-harness) [reply apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx]: Done

File: apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx (hunk 3 above)
Comment (vivek-harness) [reply apps/gitness/src/pages/run-pipeline-dialog/run-pipeline-form.tsx]: Removed console error logs

**Comment by vivek-harness** [comment]
> The alert should ideally not overlap with the dialog close button.

Added some margin

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**